src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+
+    // set the source-side flow rate
+    if (this->sourceSideDesignVolFlowRateWasAutoSized) {
+        // load-side capacity should already be set, so unless the flow rate is specified, we can set
+        // an assumed reasonable flow rate since this doesn't affect downstream components
+        Real64 DeltaT_src = 10.0;
+        // to get the source flow, we first must calculate the required heat impact on the source side
+        // First the definition of COP: COP = Qload/Power, therefore Power = Qload/COP
+        // Then the energy balance:     Qsrc = Qload + Power
+        // Substituting for Power:      Qsrc = Qload + Qload/COP, therefore Qsrc = Qload (1 + 1/COP)
+        Real64 const designSourceSideHeatTransfer = tmpCapacity * (1.0 + 1.0 / this->referenceCOP);
+        // To get the design source flow rate, just apply the sensible heat rate equation:
+        //                              Qsrc = rho_src * Vdot_src * Cp_src * DeltaT_src
+        //                              Vdot_src = Q_src / (rho_src * Cp_src * DeltaT_src)
+        tmpSourceVolFlow = designSourceSideHeatTransfer / (rhoSrc * CpSrc * DeltaT_src);
+    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate > 0.0) {
+        // given the value by the user
+        // set it directly
+        tmpSourceVolFlow = this->sourceSideDesignVolFlowRate;
+    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate == 0.0) { // LCOV_EXCL_LINE
+        // user gave a flow rate of 0
+        // protected by the input processor to be >0.0
+        // fatal out just in case
+        errorsFound = true; // LCOV_EXCL_LINE
+        ShowSevereError(state,
+                        format("Invalid condenser flow rate for EIR PLHP (name={}; entered value: {}",
+                               this->name,
+                               this->sourceSideDesignVolFlowRate)); // LCOV_EXCL_LINE
+    } else {
+        // can't imagine how it would ever get to this point
+        // just assume it's the same as the load side if we don't have any sizing information
+        tmpSourceVolFlow = tmpLoadVolFlow; // LCOV_EXCL_LINE
+    }
+
+    this->sourceSideDesignVolFlowRate = tmpSourceVolFlow;
+
+    if (errorsFound) {
+        ShowFatalError(state, "Preceding sizing errors cause program termination"); // LCOV_EXCL_LINE
+    }
+}
+
+void EIRFuelFiredHeatPump::resetReportingVariables()
+{
+    this->loadSideHeatTransfer = 0.0;
+    this->loadSideEnergy = 0.0;
+    this->loadSideOutletTemp = this->loadSideInletTemp;
+    this->powerUsage = 0.0;
+    this->powerEnergy = 0.0;
+    this->fuelUsage = 0.0;
+    this->fuelEnergy = 0.0;
+    this->sourceSideHeatTransfer = 0.0;
+    this->sourceSideOutletTemp = this->sourceSideInletTemp;
+    this->sourceSideEnergy = 0.0;
+}
+
+PlantComponent *EIRFuelFiredHeatPump::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType hp_type_of_num, const std::string &hp_name)
```

Comment:
`hp_type` is good. Please don't use `type_of` or `type_num` or `type_of_num`. Thx.

Reply:
Addressed in previous changes.